src/print_diff.ml

```diff
@@ -24,19 +24,23 @@ let print ?(skip_trailing_cr=Sys.win32) path1 path2 =
       ]
   in
   let normal_diff () =
-    match Bin.which ~path:(Env.path Env.initial) "diff" with
-    | None -> fallback ()
-    | Some prog ->
-      Format.eprintf "%a@?" Loc.print loc;
-      let* () =
-        Process.run ~dir ~env:Env.initial Strict prog
-          (List.concat
-             [ ["-u"]
-             ; if skip_trailing_cr then ["--strip-trailing-cr"] else []
-             ; [ file1; file2 ]
-             ])
-      in
-      fallback ()
+    let path, args =
+      let which prog = Bin.which ~path:(Env.path Env.initial) prog in
+      match which "git" with
+      | Some path ->
+        path, [ "diff"; "--no-index"; "--color=always"; "-u" ]
+      | None ->
+        match which "diff" with
+        | Some path ->
+          path, [ "-u" ]
+        | None ->
+          fallback ()
+    in
+    let args = if skip_trailing_cr then args @ [ "--strip-trailing-cr" ] else args in
```

Comment:
`--strip-trailing-cr` is not understood by `git diff`; the corresponding option is called `--ignore-cr-at-eol`.

Reply:
Fix in #2418